### PR TITLE
Container file resource

### PR DESCRIPTION
**Pull request by jtopjian**
This commit adds the lxd_container_file resource which can be used
to manage individual files in a container.

This depends on #112 

I can rebase afterwards or all 3 commits can be merged here.

Can also add docs once #114 is added.

### CODE REVIEW

**Comment by mjrider** [comment]
I'm wondering why you want to move the file subresource to a seperate resource
from the following 2 examples, i think the second is the most readable, and has the least surprises, so which scenario are you trying to cover here?

```hcl
resource "lxd_container" "container1" {
   name = "foobar"
   image = "images:alpine/3.5"
   profiles = ["default"]
 }
 
 resource "lxd_container_file" "file1" {
   container_name = "${lxd_container.container1.name}"
   remote  = "${lxd_container.container1.remote}"
   target_file = "/foo/bar.txt"
   content = "Hello, World!\n"
   create_directories = true
 }
```

vs
```hcl
resource "lxd_container" "container1" {
   name = "foobar"
   image = "images:alpine/3.5/amd64"
   profiles = ["default"]
   file {
     target_file = "/foo/bar.txt"
     content = "Hello, World!\n"
     create_directories = true
   }
 }
```

**Comment by jtopjian** [comment]
@mjrider See my comments in https://github.com/sl1pm4t/terraform-provider-lxd/pull/112

**Comment by sl1pm4t** [comment]
What would be the problem you see with keeping the original embedded file sub resource along side the standalone resource?
 I know it would some caveats ( the two resources compete and TF will see a diff on each run  ) but it seems to be a commonly accepted pattern in other providers. e.g google_container_cluster that includes a node_pool sub resource vs the standalone google_container_node_pool resource.

**Comment by jtopjian** [comment]
Rebased and docs added.

**Comment by mjrider** [comment]
@jtopjian what do you think about  sl1pm4t suggestion for keeping the subresource version , so users can have a choice in what they need/want

**Comment by jtopjian** [comment]
@mjrider That was my intention all along :smile:

**Comment by shantanugadgil** [comment]
FWIW, I don't mind having _only_ a separate resource.

File: lxd/shared.go [hunk 1]
```diff
@@ -127,3 +155,155 @@ func resourceLxdValidateDeviceType(v interface{}, k string) (ws []string, errors
...
+	log.Printf("[DEBUG] Successfully uploaded file %s", targetFile)
+
+	return nil
+}
+
+// containerDeleteFile will delete a file on a container.
+func containerDeleteFile(server lxd.ContainerServer, container string, targetFile string) error {
+	targetFile, err := filepath.Abs(targetFile)
+	if err != nil {
+		return fmt.Errorf("Could not santize destination target %s", targetFile)
```
Comment (sl1pm4t) [inline lxd/shared.go]: typo - `santize` -> `sanitize`

**Comment by sl1pm4t** [comment]
>  That was my intention all along 😄

Sorry for the confusion, when I read through the PR comments on my phone I somehow got the impression you were removing the embedded resource. Might be because there was another PR for deprecated a resource, and I confused the two.

Anyway it looks good to me.

**Comment by jtopjian** [comment]
No problem at all!

I detailed my plans for this in https://github.com/sl1pm4t/terraform-provider-lxd/pull/112. I should have added those notes here as well, so it's my fault for the confusion.

Thank you for catching the typo.

**Pull request merged**